### DISCUSSION TITLE
check this implementation for subscriptions over SSE

### DISCUSSION THREAD

**Discussion by ye7iaserag** [rebing/graphql-laravel#894] [Show and tell]
I tried to Implement graphQL subscription through SSE which required a minimal extension of the graphQL library
I tried to stay as close as possible to the specification in
https://spec.graphql.org/June2018/#sec-Subscription-Operation-Definitions

Please check it and tell me what you think

https://github.com/ye7iaserag/php-graphql-example

**Comment by mfn** [comment]
I took a peek look, but going through commits which mix changes in unrelated files isn't much fun.

Doesn't this need some kind of transport system to make SSE work PHPs traditional request/response? Or is this more something for Octane?

But if you think you can add this to the library incl. tests and docs, I'd anyone is welcome!

**Reply by ye7iaserag** [reply]
My solution is a bit hacky, you can check this commit for the approach https://github.com/ye7iaserag/php-graphql-example/commit/3ebd5a5be6a51c41c28be69c8657e3008f274cb3

Sorry for the unorganized commits but the project was just a hello world for using the graphql package and later deviated to an expirement for subscription implementation.

About the transport, it's http streamed response which I changed in the controller directly and from client side I used JS EventSource in graphiql which works perfectly.

It's just a proof of concept and not a full implementation, that's why I was asking for opinions.

**Comment by Abdulhadi-Zakkour** [comment]
I couldn't get this application to run on my local machine but I reused parts of your code to solve the pagination problem I [posted here ](https://github.com/rebing/graphql-laravel/discussions/959) 
Thank you!
